### PR TITLE
fix(type): Use correct object detection

### PR DESCRIPTION
**Pull request by ExE-Boss**
This makes the `type` helper only return `Object` for actual objects, ensuring that in case **ECMAScript** ever adds another primitive value, #19 won’t be required again.

It also has the added benefit of treating objects with [the `[[IsHTMLDDA]]` internal slot](https://tc39.es/ecma262/#sec-IsHTMLDDA-internal-slot) as objects, like actual **WebIDL**.

This is tested by using **V8**’s `%GetUndetecteable()` helper, which creates a function with the `[[IsHTMLDDA]]` internal slot.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -29,7 +31,16 @@ function type(V) {
...
+
+            // This will also be hit for objects with the `[[IsHTMLDDA]]` internal slot,
+            // which causes typeof to return "undefined".
+            try {
+                // %ObjProto_valueOf% acts an identity function for real objects
+                if (objValueOf.call(V) === V) {
+                    return "Object";
+                }
+            } catch (err) { /* Ignore */ }
+            return "Unknown";
```
Comment (TimothyGu) [inline lib/index.js]: It seems like we don't have to add this try-catch block, given that we have tightened up the check for Undefined values up above. "Unknown" is not a ECMA-262 value type, and we can assume that our host implements ECMA-262.

**Comment by ExE-Boss** [comment]
> whether or not we want to test for this using V8-specific features.

The only other way to do so would be to test in a web browser or **Electron**.

File: lib/index.js (hunk 1 above)
Comment (ExE-Boss) [reply lib/index.js]: This is a fallback for any potential future primitive type, though I guess I could do what ES‑Abstract does and return `undefined` in this case.

**Comment by domenic** [comment]
Since my opinion was asked, I am generally -1 on this change. It adds nontrivial code and test complexity for no gain. This is over-engineered future-proofing that hurts the current codebase. (See: YAGNI.)

I'll also note this change was unsolicited and proposed without discussion, which is not a pattern I want to encourage for PRs on the jsdom project.

**Comment by TimothyGu** [comment]
I think I agree with @domenic on this.

**Pull request closed without merging**